src/main/java/us/kilroyrobotics/RobotContainer.java

```diff
@@ -343,14 +258,51 @@ private Command alignReef(boolean leftSide) {
                 });
     }
 
-    private Timer autoLeaveTimer = new Timer();
-    private Command autoLeave =
+    /* AutoLeave Command */
+    private final Timer autoLeaveTimer = new Timer();
+    private final Command autoLeave =
             Commands.sequence(
-                            Commands.runOnce(() -> this.autoLeaveTimer.restart()),
+                            Commands.runOnce(() -> autoLeaveTimer.restart()),
                             drivetrain.applyRequest(
                                     () -> forwardStraight.withVelocityX(MetersPerSecond.of(0.5))))
                     .onlyWhile(() -> !autoLeaveTimer.hasElapsed(2));
 
+    /* Path follower */
+    private final SendableChooser<Command> autoChooser;
+
+    @SuppressWarnings("unused")
+    public RobotContainer() {
+        if (Robot.isReal() && CameraConstants.kCameraEnabled) new Camera();
+
+        NamedCommands.registerCommand(
+                "Intake Coral",
+                towerIntakeCoral
+                        .andThen(Commands.waitTime(Seconds.of(3)))
+                        .until(() -> coralIntakeMotor.isCoralDetected()));
+        NamedCommands.registerCommand("Score Coral", towerScoreCoral);
+        NamedCommands.registerCommand(
+                "GoTo L1", towerToL1.andThen(Commands.waitTime(Seconds.of(2))));
+        NamedCommands.registerCommand(
+                "GoTo L2", towerToL2.andThen(Commands.waitTime(Seconds.of(2))));
+        NamedCommands.registerCommand(
+                "GoTo L3", towerToL3.andThen(Commands.waitTime(Seconds.of(2))));
+        NamedCommands.registerCommand(
+                "GoTo L4", towerToL4.andThen(Commands.waitTime(Seconds.of(2))));
```

Comment:
Why do all of these have 2 second delays? Should we build those delays into the tower state machine?

Reply:
In autonomous, the _command_ finishes before the actual mechanisms get up there and score, causing the next path to start, so this is to make sure the coral is scored in time. This isn't an issue in teleop.

Reply:
Gotcha. For GaCo we made separate commands for autonomous that waited until the mechanism was in position before ending. Maybe we should implement that as well. We could tack on a .until() if we want to